src/cli/src/cluster/install/k8.rs

```diff
@@ -10,7 +10,7 @@ use super::*;
 pub async fn install_core(opt: InstallCommand) -> Result<(), CliError> {
     let (client, server): (TlsPolicy, TlsPolicy) = opt.tls.try_into()?;
 
-    let mut builder = ClusterInstaller::new()
+    let mut builder = ClusterInstaller::builder()
```

Comment:
Typically, in the builder pattern, builder is references not the struct it is being [constructured](https://doc.rust-lang.org/1.0.0/style/ownership/builders.html).

Reply:
A few reasons I did it this way:

- The `ClusterInstallerBuilder` acts just like a config object at the end of the day, with all the options neatly laid out. I even call it `config` inside the `ClusterInstaller`
- An API user can avoid importing the `ClusterInstallerBuilder` type because all of the necessary methods are on the `ClusterInstaller` type. E.g.

```rust
use fluvio_cluster::ClusterInstaller;
let installer = ClusterInstaller::builder()
    .with_namespace("my-namespace")
    .with_rust_log("debug")
    .build()?;
installer.install_fluvio().await?;
```

- The `build()` method on the `ClusterInstallerBuilder` is the gate where things either initialize successfully or not. For example, we can do validation on whether any options can conflict with each other. Currently, this would be the point of failure if the HelmClient or K8Client could not be initialized.
- The Builder is using the consuming calls, i.e. they take `mut self`. I didn't want to have tons of consuming calls on the same object that was actually performing the installation.
- After calling `.build()`, all of the settings are locked in. That means you couldn't accidentally call `install_sys`, then change a setting, then call `install_app` with incompatible settings.

Reply:
provisioning  helm client and K8client should be done at the time of install and should be defer to pre-check module.  Probably pre-check should can return resource as well.

Reply:
> provisioning helm client and K8client should be done at the time of install

Why is that a requirement? This is ensuring the same order or operations